Elevel/Application/Features/TestCommands/CreateTestCommand.cs

```diff
@@ -157,48 +163,118 @@ private void CreateTestQuestionsForGrammar(List<Guid> questionIds, Test test)
                         QuestionId = question
                     };
                     _context.TestQuestions.Add(testQuestion);
```

Comment:
database queries in a loop it's a bad practice. You should find solution with one query to the DB.